### DISCUSSION TITLE
Axis legends and axis title collision issue (best solution to implement)

### DISCUSSION THREAD

**Discussion by deathsh0ot** [Hexastack/eazychart#80] [General]
# Overview
### Bug :bug: : #57 
I've tried using many libraries most of which did have this collision issue to some extent, I've also tried looking for the best implementation to avoid this problem and this is what I found : **Most libraries have these collision issues, I would assume they leave it up to the developer to preprocess the data before visualizing it or provide customization options to resolve them.**

###### link to the code sample [RechartsSample](https://codesandbox.io/s/recharts-hacking-forked-ziigt0)
![image](https://user-images.githubusercontent.com/57747442/196908904-103cb64d-09a8-4bc9-ad2d-25005858d81b.png)

As we can see the tick labels themselves can collide.

###### link to the code sample: [C3jsSample](https://jsfiddle.net/1jneh8u2/2/)

![image](https://user-images.githubusercontent.com/57747442/196909948-1e5a6ba7-0a5d-467e-a485-95124b5b3495.png)
In here the tick text and axis label collide.
###### link to the code sample: [HighChartsSample](https://jsfiddle.net/pnz3sgtj/5/)
![image](https://user-images.githubusercontent.com/57747442/196910856-41491fac-3f91-4c08-a141-17165bd1123f.png)
HighCharts has no collision but I'm not sure about the implementation, long tick text may take up more precious screen space than the chart itself which seems kind of counterproductive.
###### link to the code sample [Nvd3Sample](http://jsfiddle.net/ars03dem/2/)
![image](https://user-images.githubusercontent.com/57747442/196912261-19efc992-f5ba-4077-8e59-4735b7f7c641.png)
Tick labels collide among themselves.
###### link to the code sample [Chart.jsSample](https://jsfiddle.net/r4s8z0Lj/13/)
![image](https://user-images.githubusercontent.com/57747442/196912932-79637c78-b0be-42f4-87f3-04b179eeca1f.png)
Chart.js has a similar solution to HighCharts but it appears that above a certain limit the collision occurs.

###### link to the code sample [ApexChartsSample](https://codepen.io/jbarradas/pen/PowZpJV)
![image](https://user-images.githubusercontent.com/57747442/196913674-966174fa-2694-485e-a788-f9f451d66990.png)
ApexCharts solution is that they'd truncate the axis tick text if it got over a certain length and the full text would be displayed as a tooltip when hovered over.
### Conclusion :end: 
After trying out these libraries ApexCharts solution was the most appropriate in my opinion.
I created this discussion to share my findings and to get more opinions on this matter, so if you think you have a better solution or disagree on some parts please feel free to share.

**Comment by marrouchi** [comment]
Looks like the ApexCharts way of solving this is the best. @deathsh0ot  I would suggest we start on solving this only for the vertical axis.

**Reply by deathsh0ot** [reply]
Alright I'm on it.
